### PR TITLE
Created review spending cap component

### LINKED ISSUE #15212: Create review spending cap component
## Resources
[Figma Design file](https://www.figma.com/file/n0xMnZqQz4w9Sfa3ZdhWUi/Consentful-experience-for-token-allowances?node-id=2007%3A14202)

## Acceptance Criteria

1. TBD
2.

2. Clicking on `edit`, takes the user back to the first token allowance screen (a.k.a. Set spending cap screen)

### PR DESCRIPTION
**Pull request by VSaric**
## Explanation

Created a review spending cup component. This component can't yet be added to the app. Instead, I added a storybook story that displays this component. This component received at least the following props:

1. tokenName (type: string)
2. currentTokenBalance (type: number)
3. tokenValue (type: number)
4. onEdit (type: function)

## More Information

* Fixes #15212 

## Screenshots/Screencaps

https://user-images.githubusercontent.com/92527393/185385264-26d9604c-f76f-4149-919a-66034606ec27.mov

## Manual Testing Steps

This component can be tested by running storybook (`yarn storybook`) and then go to the `COMPONENTS` tab and open in `UI/ReviewSpendingCup`. For testing this component you can see the Acceptance Criteria in the ticket: https://github.com/MetaMask/metamask-extension/issues/15212.

### CODE REVIEW

**Comment by mirjanaKukic** [comment]
Verified by QA

File: ui/components/ui/review-spending-cap/index.scss [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+      &__question-icon {
+        color: var(--color-icon-muted);
+      }
+    }
+  }
+
+  &__heading-detail {
+    flex-grow: 1;
+    align-self: center;
```
Comment (brad-decker) [inline ui/components/ui/review-spending-cap/index.scss:22]: @georgewrmarshall marking this as a request for extending Box props.

**Review by brad-decker — APPROVED** [review]
LGTM great work @VSaric

**Review by darkwing — APPROVED** [review]

File: ui/components/ui/review-spending-cap/index.scss (hunk 1 above)
Comment (georgewrmarshall) [reply ui/components/ui/review-spending-cap/index.scss:22]: Noted. https://github.com/MetaMask/metamask-extension/issues/15685

**Pull request merged**